src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy

```diff
@@ -241,9 +244,29 @@ class ProtobufPlugin implements Plugin<Project> {
       sourceSetNames.each { sourceSetName ->
         def extractProtosTask = maybeAddExtractProtosTask(sourceSetName)
         generateProtoTask.dependsOn(extractProtosTask)
+      }
 
-        def extractIncludeProtosTask = maybeAddExtractIncludeProtosTask(sourceSetName)
+      // Extract proto file from the compile classpath configuration exposed in Android Gradle plugin 2.5.  Revert to
+      // original behavior if the configuration is not found.
+      Configuration compileConfiguration = variant.hasProperty("compileConfiguration") ? variant.compileConfiguration : null
+      if (compileConfiguration != null) {
```

Comment:
`protobuf-gradle-plugin/testProjectAndroid/build.gradle` is still using `com.android.tools.build:gradle:2.2.0-beta1`. This new branch is not tested. You will need some work on the test code (`ProtobufAndroidPluginTest` and its friends) to test both the old and the new Android plugins.